File: backend/Services/Parsers/SemantleJuniorParser.cs
using System.Text.RegularExpressions;
using backend.Models;

namespace backend.Services.Parsers
{
    public class SemantleJuniorParser : SemantleParser
    {
        private ILogger<SemantleJuniorParser> _logger;

        public SemantleJuniorParser(ILogger<SemantleJuniorParser> logger) : base(logger)
        {
            _logger = logger;
        }

        public override string GameName => "Semantle Junior";
        public override string Url => "https://semantle.com/junior";
    }
}

File: lambdas/src/parsers/semantlejunior-parser.ts
import { SemantleParser } from './semantle-parser';

export class SemantleJuniorParser extends SemantleParser {
  override readonly gameName = 'Semantle Junior';
  override readonly url = 'https://semantle.com/junior';
  
  // Override the regex to match "Semantle Junior" specifically
  protected override readonly parser = new RegExp(
    `Semantle Junior #\\d+.*?(?<completed>✅|❌).*?(?<score>\\d+) Guesse?s?.*?💡 (?<hints>\\d+) Hints?`,
    's'
  );
}
